### PR TITLE
Add catcol as colour name to use the category colour for the icon

### PR DESCRIPTION
**Pull request by rogercreagh**
Allow catcol as colour specifier to select the category badge colour.

eg "test,hands-helping,catcol,partial"

Most useful with the partial name match so that, in the example, the same icon can be used for all categories with "test" in their slug but they will each pick up their own category colour rather than all using the same. Also provides a quick alternative to having to select a colour for a one-off icon (exact match) - simply use the category colour.

This will also be useful with a proposed further new option to have the icon replace the category stripe whilst maintaining the category colour hint - to be subject of a future PR

### CODE REVIEW

**Review by pmusaraj — CHANGES_REQUESTED** [review]
This looks like a useful change, thanks! I would prefer to call the special option "categoryColor", it's easier to understand than "catcol".

Can you also remove the extra files in the commit? Looks like they are coming from Eclipse.

**Comment by rogercreagh** [comment]
Whoops - sorry I though Eclipse should have filtered them out. They should be gone now and also I changed catcol to categoryColour as you suggested.
I haven't changed the version number in the about.json (it doesn't seem to be there) and also when I tried the current (prior to this PR) on a Discourse 2.5 install it didn't seem to work, although fine on 2.6. Should there be a minimum discourse version specified in there?

File: javascripts/discourse/initializers/category-icons.js [hunk 1]
```diff
@@ -107,7 +107,8 @@ export default {
         /// Add custom category icon from theme settings
         let iconItem = getIconItem(category.slug);
         if (iconItem) {
-          let itemColor = iconItem[2] ? `style="color: ${iconItem[2]}"` : "";
+        let itemColor = iconItem[2] == 'categoryColour' ? `style="color: #${color}"` : iconItem[2] ? `style="color: ${iconItem[2]}"` : "";
```
Comment (ZogStriP) [inline javascripts/discourse/initializers/category-icons.js:110]: Looks like the indentation is off here.

File: javascripts/discourse/initializers/category-icons.js [hunk 2]
```diff
@@ -107,7 +107,8 @@ export default {
...
+        ///  let itemColor = iconItem[2] ? `style="color: ${iconItem[2]}"` : "";
```
Comment (ZogStriP) [inline javascripts/discourse/initializers/category-icons.js:111]: Can you remove that comment?

File: javascripts/discourse/initializers/category-icons.js (hunk 1 above)
Comment (rogercreagh) [reply javascripts/discourse/initializers/category-icons.js:110]: done

File: javascripts/discourse/initializers/category-icons.js (hunk 2 above)
Comment (rogercreagh) [reply javascripts/discourse/initializers/category-icons.js]: Done

File: .gitignore [hunk 3]
```diff
@@ -1 +1,71 @@
-.discourse-site
\ No newline at end of file
+.discourse-site
```
Comment (pmusaraj) [inline .gitignore:1]: Sorry to keep pushing back on this @rogercreagh, but this `.gitignore` file showed up again...

**Pull request merged**